src/hooks/useUsers.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { User } from '../types';
+import * as usersService from '../api/users';
+
+export const useUsers = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadUsers = () => {
+    setErrorMessage('');
+
+    usersService
+      .getUsers()
+      .then(res => {
+        const preparedUsers = res.map(user => {
+          const { id, name, email, phone } = user;
+
+          return {
+            id,
+            name,
+            email,
+            phone,
+          };
+        });
+
+        setUsers(preparedUsers);
+      })
+      .catch(() => setErrorMessage('Something went wrong'));
```

Comment:
Consider providing more detailed error messages or logging the error to help with debugging. The current message 'Something went wrong' is generic and may not provide enough information for troubleshooting.